Update passes in quant2_int8_mkldnn_pass

wozna opened the pull request:
<!-- Demo: https://github.com/PaddlePaddle/Paddle/pull/24810 -->
### PR types
<!-- One of [ New features | Bug fixes | Function optimization | Performance optimization | Breaking changes | Others ] -->
Bug fixes
### PR changes
<!-- One of [ OPs | APIs | Docs | Others ] -->
Others 
### Describe
<!-- Describe what this PR does -->
This PR updates passes that are applied to the graph during quant2_int8_mkldnn_pass. It turns out that many passes were added to CpuPassStrategy and EnableMkldnn passes, but they weren't added to mentioned python script. I used the same order of the passes that is presented in `paddle_pass_builder.cc`.

Commit:
Upadate pass in quant2_int8_mkldnn_pass

lidanqing-vv commented:
@wozna @pmajchrzak Please verify this PR will not cause our existing daily CI int8 models accuracy and performance drop. Thanks!

sfraczek reviewed (dismissed):
LGTM

lidanqing-vv commented:
@wozna Hi could you or ask Pior to upload spreadsheet showing this PR does not cause any accuracy drop or performance drop on exisiting int8 modes CI

Commit:
Back to the previous scale_matmul order

Commit:
Change place of cpu_quantize_placement_pass

wozna commented:
I can confirm that all recent changes don't cause any accuracy change or performance change in any model from our CI. Before I had a problem with `scale_matmul_fuse_pass`. I couldn't understand why this pass has to be in `_quantize_fp32_graph` not in the `_optimize_fp32_graph`. It turned out that we use a scale operator to gather scale for matmul and when I changed the place of `scale_matmul_fuse_pass`, 12 matmuls weren't quantized because the scale for input was missing. This change improved a little accuracy on Ernie from 0.791165 to 0.8000 but caused a performance drop of 1.5%.  That's why I returned to the previous order, but I moved  `cpu_quantize_placement_pass`, after `scale_matmul` and `reshape_transpose_matmul/V2` passes.

wozna commented:
@lidanqing-intel @sfraczek Could you please repeat your review?

sfraczek reviewed (approved):
LGTM.

lidanqing-intel reviewed (approved):
LGTM

lidanqing-vv commented:
@baoachun Could you please review this PR. Thanks

lidanqing-vv commented:
@baoachun Hi could you please approve this PR? Some models from BML team still need to use save_quant_model.py solution for now. So we still need to maintain this save_quant_model.py. Thanks

baoachun reviewed (approved):
LGTM
